### PR TITLE
#396: rename inactive device deletion config

### PR DESCRIPTION
**Pull request by fbuetler**
Fixes https://github.com/freifunkMUC/wg-access-server/issues/396

### CODE REVIEW

File: cmd/serve/main.go [hunk 1]
```diff
@@ -45,8 +45,8 @@ func Register(app *kingpin.Application) *servecmd {
 	cli.Flag("external-host", "The external origin of the server (e.g. https://mydomain.com)").Envar("WG_EXTERNAL_HOST").StringVar(&cmd.AppConfig.ExternalHost)
 	cli.Flag("storage", "The storage backend connection string").Envar("WG_STORAGE").Default("memory://").StringVar(&cmd.AppConfig.Storage)
 	cli.Flag("disable-metadata", "Disable metadata collection (i.e. metrics)").Envar("WG_DISABLE_METADATA").Default("false").BoolVar(&cmd.AppConfig.DisableMetadata)
-	cli.Flag("disable-inactive", "Disable inactive device deletion").Envar("WG_DISABLE_INACTIVE").Default("false").BoolVar(&cmd.AppConfig.DisableInactive)
-	cli.Flag("inactive-duration", "Duration after inactive device are deleted").Envar("WG_INACTIVE_DURATION").Default("4380h").DurationVar(&cmd.AppConfig.InactiveDuration)
+	cli.Flag("enable-inactive-device-deletion", "Enable inactive device deletion").Envar("WG_ENABLE_INACTIVE_DEVICE_DELETION").Default("false").BoolVar(&cmd.AppConfig.EnableInactiveDeviceDeletion)
+	cli.Flag("inactive-device-grace-period", "Duration after inactive device are deleted").Envar("WG_INACTIVE_DEVICE_GRACE_PERIOD").Default("4380h").DurationVar(&cmd.AppConfig.InactiveDeviceGracePeriod)
```
Comment (nwaldispuehl) [inline cmd/serve/main.go:49]: The `"4380h"` are kind of hard to parse. I am aware that there is no day (`d`) with these durations, but could we introduce a variable for it, e.g.:
```
var HALF_YEAR = "4380h"  
```
or `SIX_MONTHS` or something?

File: docs/2-configuration.md [hunk 2]
```diff
@@ -27,33 +27,35 @@ The format for specifying multiple values for options that allow it is:
...
+| `WG_ENABLE_INACTIVE_DEVICE_DELETION` | `--enable-inactive-device-deletion` | `enableInactiveDeviceDeletion` |          | `false`                                      | Enable/Disable the automatic deletion of inactive devices.                                                                                                                                                                                                         |
+| `WG_INACTIVE_DURATION`               | `--inactive-duration`               | `inactiveDuration`             |          | `4380h`                                      | The duration after which inactive devices are automatically deleted, if automatic deletion is enabled. A device is inactive if it has not been connected to the server for longer than the inactive duration. The duration format is the go duration string format |
```
Comment (nwaldispuehl) [inline docs/2-configuration.md:41]: This also needs to be adapted to the new duration name -- `WG_INACTIVE_DEVICE_GRACE_PERIOD`.

File: internal/devices/inactive.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func checkAndRemove(d *DeviceManager, inactiveDuration time.Duration) {
 			elapsed = time.Since(*dev.LastHandshakeTime)
 		}
 
-		if elapsed > inactiveDuration {
+		if elapsed > inactiveDeviceGracePeriod {
 			logrus.Debug("deleting inactive device")
```
Comment (nwaldispuehl) [inline internal/devices/inactive.go:38]: Can we add the owner and name of the device to the log statement?

File: internal/services/server_service.go [hunk 4]
```diff
@@ -56,22 +56,22 @@ func (s *ServerService) Info(ctx context.Context, req *proto.InfoReq) (*proto.In
...
-		IsAdmin:          user.Claims.IsAdmin(),
-		AllowedIps:       allowedIPs(s.Config),
-		DnsEnabled:       s.Config.DNS.Enabled,
-		DnsAddress:       dnsAddress,
-		Filename:         s.Config.Filename,
-		ClientConfigDnsServers:        clientConfigDnsServers(s.Config),
-		ClientConfigDnsSearchDomain:   s.Config.ClientConfig.DNSSearchDomain,
+		HostVpnIp:                   hostVPNIP,
+		MetadataEnabled:             !s.Config.DisableMetadata,
+		InactiveEnabled:             s.Config.EnableInactiveDeviceDeletion,
```
Comment (nwaldispuehl) [inline internal/services/server_service.go:66]: I assume the keys are hard to change?

**Review by nwaldispuehl — CHANGES_REQUESTED** [review]
Thanks! :blush:

**Review by GoliathLabs — APPROVED** [review]

File: cmd/serve/main.go (hunk 1 above)
Comment (fbuetler) [reply cmd/serve/main.go:49]: We can do the same as the `time` package:

```go
const (
	Day  time.Duration = 24 * time.Hour
	Year                        = 365 * Day
)
```

and set the default to 1 year, because 6 month is kinda hard to represent and the default was chosen arbitrarily anyway.

File: internal/services/server_service.go (hunk 4 above)
Comment (fbuetler) [reply internal/services/server_service.go:66]: a little bit but not too much 🙃  gonna take a look

Comment (nwaldispuehl) [reply internal/services/server_service.go]: Nice, thanks.
It is to be hoped that nobody used the feature so far... :smile:

**Comment by fbuetler** [comment]
ready for another round 🙂

**Review by nwaldispuehl — APPROVED** [review]
